pkg/apis/pipeline/v1/pipeline_types_test.go

```diff
@@ -129,6 +285,54 @@ func TestPipelineTask_ValidateRefOrSpec(t *testing.T) {
 	}
 }
 
+func TestPipelineTask_ValidateRefOrSpec_IncompatibleAPIVersions(t *testing.T) {
+	tests := []struct {
+		name            string
+		requiredVersion string
+		pt              PipelineTask
+	}{
+		{
+			name:            "pipelineRef requires alpha version",
+			requiredVersion: config.AlphaAPIFields,
+			pt: PipelineTask{
+				PipelineRef: &PipelineRef{},
+			},
+		}, {
+			name:            "pipelineSpec requires alpha version",
+			requiredVersion: config.AlphaAPIFields,
+			pt: PipelineTask{
+				PipelineSpec: &PipelineSpec{},
+			},
+		},
+	}
+
+	versions := []string{config.AlphaAPIFields, config.BetaAPIFields, config.StableAPIFields}
+	for _, test := range tests {
+		for _, version := range versions {
+			name := fmt.Sprintf("%s with version %s", test.name, version)
+			t.Run(name, func(t *testing.T) {
+				featureFlags, _ := config.NewFeatureFlagsFromMap(map[string]string{
+					"enable-api-fields": version,
+				})
+				cfg := &config.Config{
+					FeatureFlags: featureFlags,
+				}
+
+				ctx := config.ToContext(context.Background(), cfg)
+				err := test.pt.validateRefOrSpec(ctx)
```

Comment:
We have helper function to set API fields and other feature flags:

https://github.com/tektoncd/pipeline/blob/2299b1551299b976e6406f1b97e046ff05e9ae39/pkg/apis/config/testing/featureflags.go#L43C1-L45C2

Reply:
thanks @QuanZhang-William, replaced with the helper function.